backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -104,61 +115,88 @@ func PeriodicallyRefreshCache(s *Server) {
 	go func() {
 		for range ticker.C {
 			s.logger.Info("Refreshing RTDS cache")
-			refreshCache(s.ctx, s.storer, &cacheWrapperV2{s.snapshotCacheV2}, s.rtdsLayerName, s.logger)
-			refreshCache(s.ctx, s.storer, &cacheWrapperV3{s.snapshotCacheV3}, s.rtdsLayerName, s.logger)
+			refreshCache(s.ctx, s.storer, &cacheWrapperV2{s.snapshotCacheV2}, s.rtdsLayerName, s.ingressPrefix, s.egressPrefix, s.logger)
+			refreshCache(s.ctx, s.storer, &cacheWrapperV3{s.snapshotCacheV3}, s.rtdsLayerName, s.ingressPrefix, s.egressPrefix, s.logger)
 		}
 	}()
 }
 
 func refreshCache(ctx context.Context, storer experimentstore.Storer, snapshotCache cacheWrapper, rtdsLayerName string,
-	logger *zap.SugaredLogger) {
-	allExperiments, err := storer.GetExperiments(ctx, "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig", experimentation.GetExperimentsRequest_STATUS_RUNNING)
+	ingressPrefix string, egressPrefix string, logger *zap.SugaredLogger) {
+	allRunningExperiments, err := storer.GetExperiments(ctx, "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig", experimentation.GetExperimentsRequest_STATUS_RUNNING)
 	if err != nil {
 		logger.Errorw("Failed to get data from experiments store", "error", err)
 
 		// If failed to get data from DB, stop all ongoing faults.
-		allExperiments = []*experimentation.Experiment{}
+		allRunningExperiments = []*experimentation.Experiment{}
 	}
 
+	// INGRESS Faults
 	// Group faults by upstream cluster
 	upstreamClusterFaultMap := make(map[string][]*experimentation.Experiment)
-	for _, experiment := range allExperiments {
+	for _, experiment := range allRunningExperiments {
 		testConfig := &serverexperimentation.TestConfig{}
-		if !isFaultTest(experiment, testConfig) {
-			continue
+		if isFaultTest(experiment, testConfig) && testConfig.FaultInjectionType == serverexperimentation.FaultInjectionType_FAULTINJECTIONTYPE_INGRESS {
+			clusterPair := testConfig.GetClusterPair()
+			upstreamClusterFaultMap[clusterPair.UpstreamCluster] =
+				append(upstreamClusterFaultMap[clusterPair.UpstreamCluster], experiment)
+		}
+	}
+
+	// EGRESS Faults
+	// Group faults by downstream cluster
+	downstreamClusterFaultMap := make(map[string][]*experimentation.Experiment)
+	for _, experiment := range allRunningExperiments {
+		testConfig := &serverexperimentation.TestConfig{}
+		if isFaultTest(experiment, testConfig) && testConfig.FaultInjectionType == serverexperimentation.FaultInjectionType_FAULTINJECTIONTYPE_EGRESS {
```

Comment:
can we check `testConfig.FaultInjectionType == serverexperimentation.FaultInjectionType_FAULTINJECTIONTYPE_EGRESS` - it's way less cheaper compared to `isFaultTest(experiment, testConfig)` - the same comments applies to other similar checks in this file

Reply:
Not sure what you mean by `way less cheaper`? 
But I have refactored the code and this is now updated.